File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/interpreter/lab/Literal.java
// Literal.java

import java.io.PrintWriter;

public class Literal extends Terminal
{
	public Literal(int aValue)
	{
		setValue(aValue);
	}

	public int getNumberOfUsesOfVariable(String aName)
	{
		int result = 0;

		return result;
	}

	public void output(PrintWriter aPrintWriter)
	{
		aPrintWriter.print(getValue());
	}

	private void setValue(int aValue)
	{
		myValue = aValue;
	}

	public int getValue()
	{
		return myValue;
	}

	private int myValue;
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/interpreter/solution/literal.cpp
// literal.cpp

#include <iostream>
#include <string>
#include "dp_map.h"
class expression;
#include "literal.h"

literal::literal(int a_value)
{
	set_value(a_value);
}

int literal::get_number_of_uses_of_variable(const std::string & a_name) const
{
	int result(0);

	return result;
}

double literal::get_value(const dp_map<std::string, double> & a_map) const
{
	return get_value();
}

void literal::output(std::ostream & a_ostream) const
{
	a_ostream << get_value();
}

void literal::set_value(int a_value)
{
	my_value = a_value;
}

int literal::get_value() const
{
	return my_value;
}
